k-distribution/src/main/scripts/lib/pyk/ktool.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+
+from .kastManip import *
+
+class KPrint:
+    """Given a kompiled directory, build an unparser for it.
+    """
+
+    def __init__(self, kompiledDirectory):
+        self.kompiledDirectory = kompiledDirectory
+        self.definition        = readKastTerm(self.kompiledDirectory + '/compiled.json')
+        self.symbolTable       = buildSymbolTable(self.definition, opinionated = True)
+
+    def prettyPrint(self, kast):
+        """Given a KAST term, pretty-print it using the current definition.
+
+        -   Input: KAST term in JSON.
+        -   Output: Best-effort pretty-printed representation of the KAST term.
+        """
+        return prettyPrintKast(kast, self.symbolTable)
+
+class KProve(KPrint):
+    """Given a kompiled directory and a main file name, build a prover for it.
+    """
+
+    def __init__(self, kompiledDirectory, mainFileName, useDirectory = None):
+        super(KProve, self).__init__(kompiledDirectory)
+        self.directory      = '/'.join(self.kompiledDirectory.split('/')[0:-1])
+        self.useDirectory   = self.directory + '/kprove' if useDirectory is None else useDirectory
+        if not os.path.exists(self.useDirectory):
+            os.makedirs(self.useDirectory)
+        self.mainFileName   = mainFileName
+        self.prover         = [ 'kprovex' ]
+        self.proverArgs     = [ ]
+        with open(self.kompiledDirectory + '/backend.txt', 'r') as ba:
+            self.backend    = ba.read()
+        with open(self.kompiledDirectory + '/mainModule.txt', 'r') as mm:
+            self.mainModule = mm.read()
```

Comment:
Do we have tests for these?
In case we change the contents of those files.

Reply:
I modified the testing harness to use `kpyk prove ...` (https://github.com/kframework/k/pull/2338/commits/3a534ab0dc9da0c269dfa493835b7bbd3871b417).

`kpyk prove ...` is really not intended to be used directly, it's just a wrapper that's there so we can have tests like this. But because `kpyk prove` builds a `KProve` object, that test will fail if those files aren't present.